packages/core/test/rehydration.test.ts

```diff
@@ -179,6 +179,44 @@ describe('rehydration', () => {
     expect(rehydratedActor.getSnapshot().value).toBe('c');
   });
 
+  it('should be able to restart and subscribe to a stopped rehydrated child', async () => {
+    const spy = jest.fn();
+    const machine = createMachine({
+      initial: 'a',
+      states: {
+        a: {
+          invoke: {
+            src: fromPromise(() => Promise.resolve(11)),
+            onDone: 'b'
+          },
+          on: {
+            NEXT: 'c'
+          }
+        },
+        b: {},
+        c: {}
+      }
+    });
+
+    const actor = createActor(machine).start();
+    actor.stop();
+    const persistedState = actor.getPersistedSnapshot();
```

Comment:
You should persist the snapshot before stopping the `actor`.
```suggestion
    const persistedState = actor.getPersistedSnapshot();
    actor.stop();
```